### PR TITLE
Increase ballerina runtime max yield depth size

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/34660

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by warunalakshitha** [comment]
When yield after a considerable number of function calls there is a possibility of getting StackOverflow. as per https://github.com/ballerina-platform/ballerina-lang/issues/34660. I have temporarily increased the size to 256. Proper fix would be to increase the stack size on demand.

**Comment by warunalakshitha** [comment]
This is a tempory fix and we will provide proper fix with https://github.com/ballerina-platform/ballerina-lang/issues/34671

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java [hunk 1]
```diff
@@ -86,6 +86,8 @@
     private ListenerRegistry listenerRegistry;
     private AtomicReference<ItemGroup> objectGroup = new AtomicReference<>();
 
+    private static final int BALLERINA_MAX_STACK_SIZE = 256;
```
Comment (manuranga) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java:89]: Shall we call this `BALLERINA_MAX_YIELD_DEPTH`. Max limit for stack size is governed by java, we only limit the stack size of the sleeping thread, ie max depth a yield can happen.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MainMethodGen.java [hunk 2]
```diff
@@ -396,7 +396,7 @@ private void genSubmitToScheduler(String initClass, MethodVisitor mv, String lam
         storeFuture(indexMap, mv, futureVar);
         mv.visitFieldInsn(GETFIELD , FUTURE_VALUE , STRAND,
                          GET_STRAND);
-        mv.visitIntInsn(BIPUSH, 100);
+        mv.visitIntInsn(BIPUSH, 256);
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MainMethodGen.java:399]: Better to move this to a constant as well.

**Review by manuranga — DISMISSED** [review]

**Comment by manuranga** [comment]
This issue occurring only now due to #34672

**Review by HindujaB — APPROVED** [review]
LGTM

**Pull request merged**